PSMDB-1091 Extended the step about using custom db data and log paths (#781)
percona/psmdb-docs: docs/install/upgrade-from-mongodb.md

An in-place upgrade is done by keeping the existing data in the server and replacing the MongoDB binaries. Afterwards, you restart the `mongod` service with the same `dbpath` data directory.

An in-place upgrade is suitable for most environments except the ones that use ephemeral storage and/or host addresses.

## Procedure

!!! note

    MongoDB creates a user that belongs to two groups, which is a potential security risk. This is fixed in Percona Server for MongoDB: the user is included only in the `mongod` group. To avoid problems with current MongoDB setups, existing user group membership is not changed when you migrate to Percona Server for MongoDB.  Instead, a new `mongod` user is created during installation, and it belongs to the `mongod` group.

This procedure describes an in-place upgrade of a `mongod` instance. If you are using data at rest encryption, refer to the [Upgrading to Percona Server for MongoDB with data at rest encryption enabled](upgrading-to-percona-server-for-mongodb-with-data-at-rest-encryption-enabled) section.

!!! important 

    Before starting the upgrade, we recommend to perform a full backup of your data.

=== "Upgrade on Debian and Ubuntu"

>      1. Back up the configuration file:

Reviewer: ```suggestion
     1. Save current configuration file as a backup:
```

         ```{.bash data-prompt="$"}

>          $ sudo cp /etc/mongod.conf /etc/mongod.conf.bkp

Reviewer: ```suggestion
         $ sudo mv /etc/mongod.conf /etc/mongod.conf.bkp
```

           mongodb-org-tools
         ```

     5. [Install Percona Server for MongoDB](apt.md)

     6. Verify that the configuration file includes correct options:

>          * Add the required configuration from the backup configuration file (`/etc/mongod.conf`) like custom dbPath/system log path, additional security/replication or sharding options. 

Reviewer: ```suggestion
         * Copy the required configuration options from the backup configuration file (`/etc/mongod.conf.bkp`) to the current one (`/etc/mongod.conf`) like custom dbPath/system log path, additional security/replication or sharding options.
```

         * Make sure that the `mongod` user has access to your custom paths. If not, provide it as follows:

            ```{.bash data-prompt="$"}
            $ sudo chown -R mongod:mongod <custom-dbPath>
            $ sudo chown -R mongod:mongod <custom-systemLog.path>
            ```

         * Make sure the configuration file includes the following configuration:

            ```yaml
            processManagement:
               fork: true
               pidFilePath: /var/run/mongod.pid
            ```

            **Troubleshooting tip**: The `pidFilePath` setting in `mongod.conf` must match the `PIDFile` option in the `systemd mongod` service unit. Otherwise, the service will kill the `mongod` process after a timeout.

>      7. Start the `mongod` service:

Reviewer: ```suggestion
     7. Restart the `mongod` service:
```

           mongodb-org-tools
         ```

     5. [Install Percona Server for MongoDB](apt.md)

     6. Verify that the configuration file includes correct options:

>          * Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf`) to the current one `/ect/mongodb.conf`. 

Reviewer: ```suggestion
         * Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf.bkp`) to the current one `/etc/mongodb.conf`. 
```

>          * Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf`) to the current one `/etc/mongodb.conf`. 

Reviewer: ```suggestion
         * Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf.bkp`) to the current one `/etc/mongodb.conf`. 
```

         * Make sure that the `mongod` user has access to your custom paths. If not, provide it as follows:

            ```{.bash data-prompt="$"}
            $ sudo chown -R mongod:mongod <custom-dbPath>
            $ sudo chown -R mongod:mongod <custom-systemLog.path>
            ```

         * Make sure the configuration file includes the following configuration:

            ```yaml
            processManagement:
               fork: true
               pidFilePath: /var/run/mongod.pid
            ```

            **Troubleshooting tip**: The `pidFilePath` setting in `mongod.conf` must match the `PIDFile` option in the `systemd mongod` service unit. Otherwise, the service will kill the `mongod` process after a timeout.

     7. Restart the `mongod` service:

         ```{.bash data-prompt="$"}

>          $ sudo systemctl start mongod

Reviewer: ```suggestion
         $ sudo systemctl restart mongod
```

         mongodb-org-mongos-5.0.2-1.el8.x86_64 \
         ```
     
     4. [Install Percona Server for MongoDB](yum.md)

    5. Verify that the configuration file includes correct options:

>          * When you remove old packages, your existing configuration file is saved as `/etc/mongod.conf.rpmsave`. Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf`) to the current one `/ect/mongodb.conf`.

Reviewer: ```suggestion
         * When you remove old packages, your existing configuration file is saved as `/etc/mongod.conf.rpmsave`. Copy the required configuration options like custom dbPath/system log path, additional security/replication or sharding options from the backup configuration file (`/etc/mongod.conf.rpmsave`) to the current one `/etc/mongod.conf`.
```

         * Make sure that the `mongod` user has access to your custom paths. If not, provide it as follows:

            ```{.bash data-prompt="$"}
            $ sudo chown -R mongod:mongod <custom-dbPath>
            $ sudo chown -R mongod:mongod <custom-systemLog.path>
            ```

         * Make sure the configuration file includes the following configuration:

            ```yaml
            processManagement:
               fork: true
               pidFilePath: /var/run/mongod.pid
            ```

            **Troubleshooting tip**: The `pidFilePath` setting in `mongod.conf` must match the `PIDFile` option in the `systemd mongod` service unit. Otherwise, the service will kill the `mongod` process after a timeout.

    6. Restart the `mongod` service:

        ```{.bash data-prompt="$"}

>         $ sudo systemctl start mongod

Reviewer: ```suggestion
        $ sudo systemctl restart mongod
```
